### PR TITLE
feat(react): update search result has result counts for each entities that has result

### PR DESCRIPTION
**Pull request by topwebtek7**
## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

image

### CODE REVIEW

File: datahub-web-react/src/app/useGetAuthenticatedUser.tsx [hunk 1]
```diff
@@ -6,7 +6,7 @@ import { useGetUserQuery } from '../graphql/user.generated';
  * Fetch a CorpUser object corresponding to the currently authenticated user.
  */
 export function useGetAuthenticatedUser() {
-    const userUrn = Cookies.get(CLIENT_AUTH_COOKIE);
+    const userUrn = process.env.JEST_WORKER_ID !== undefined ? 'urn:li:corpuser:1' : Cookies.get(CLIENT_AUTH_COOKIE);
```
Comment (gabe-lyons) [inline datahub-web-react/src/app/useGetAuthenticatedUser.tsx:9]: let's not include anything related to tests in this method- if you need to mock this it should be mocked in the test file itself.

File: datahub-web-react/src/app/search/SearchPage.tsx [hunk 2]
```diff
@@ -94,15 +152,20 @@ export const SearchPage = () => {
...
                             tab={
                                 <>
-                                    {entityRegistry.getIcon(t, 16, IconStyleType.TAB_VIEW)}
-                                    <StyledTab>{entityRegistry.getCollectionName(t)}</StyledTab>
+                                    {entityRegistry.getIcon(type, 16, IconStyleType.TAB_VIEW)}
+                                    <StyledTab>{entityRegistry.getCollectionName(type)}</StyledTab>
+                                    {resultCounts[type] ? (
+                                        <StyledNumberInTab>{` (${countFormatter(
+                                            resultCounts[type] || 0,
+                                        )})`}</StyledNumberInTab>
```
Comment (gabe-lyons) [inline datahub-web-react/src/app/search/SearchPage.tsx:164]: let's remove the parentheses now that the text is gray

File: datahub-web-react/src/app/search/SearchPage.tsx [hunk 3]
```diff
@@ -54,6 +68,45 @@ export const SearchPage = () => {
...
+                    resultCount += allSearchResultsByType[key].data?.search?.total;
+                    counts[key as EntityType] = allSearchResultsByType[key].data?.search?.total || 0;
+                }
+            });
+
+            analytics.event({
+                type: EventType.SearchResultsViewEvent,
+                query,
+                total: resultCount,
+            });
```
Comment (gabe-lyons) [inline datahub-web-react/src/app/search/SearchPage.tsx:104]: please move the analytics event into its own useEffect- it is a bit confusing that this is a side effect of the use memo

**Review by gabe-lyons — APPROVED** [review]
LGTM!

**Review by shirshanka — APPROVED** [review]
LGTM!

**Pull request merged**